lib/src/sync.js

```diff
@@ -82,26 +86,37 @@ module.exports = {
 
   async getAll(type, options) {
     let page = 1;
-    let res = await this.getPage(type, page, options);
-    let all =
-      res.data[type].constructor === Object ? Object.values(res.data[type]) : res.data[type];
-    let lastPage = Math.ceil(res.total / 25);
+    let initialResponse = await this.getPage(type, page, options);
+    let allItems =
+      initialResponse.data[type].constructor === Object
+        ? Object.values(initialResponse.data[type])
+        : initialResponse.data[type];
+    let lastPage = Math.ceil(initialResponse.total / 25);
 
-    while (page < lastPage) {
-      page++;
-      res = await this.getPage(type, page, options);
-      res.data[type].forEach((item) => {
-        all.push(item);
-      });
+    const fetchPromises = [];
```

Comment:
@demirgazetic If we want the loop to break _based on a condition_ other than the number of times it runs, we should use `while`. That's the approach we took before your PR, the sequential fetching.

But I would like to ask you if `for` is purposely setting the strict rate limit control.
It's also a good idea to optimize the speed as much as possible.
From what I saw, Gatsby's side does not officially have a solution for the slow `source and transform nodes`, and it is in the user's hands to optimize it, sadly, regardless of the pain for Gatsby users.

Also, when declaring `getPage`, it’s `getPage(type, page, options)`, but when calling it inside of `getAll`, the param `page` is changed to `p`.
Do you think it'll be more straightforward to keep the same naming convention with `while` if what I asked was not intentional for your concurrent fetching approach?

Please feel free to correct me or add any additional information to help us complete reviewing your PR👍

Reply:
@schabibi1  thank you for the feedback. The shift from a while loop to a for loop was primarily to optimize the speed of fetching by using a concurrent fetching approach with Promise.all. During testing, this approach reduced the fetch time significantly from 71-140 seconds with sequential fetching to 13-30 seconds with concurrent fetching, even when handling a few thousand stories.

However, the question regarding the intention behind using `for` for strict rate limit control is valid. The use of for in this context is not specifically meant to enforce strict rate limit control, but rather to facilitate the collection of promises for concurrent fetching.  

 Regarding the get `page` and the `p` I will follow up with a new PR to update this.

Reply:
@demirgazetic Thank you for the details! After I reviewed it, I noticed that `Promise.all` with `while` wouldn't work. It's clear now why you used `for`. Also, a few thousand stories to handle in 13-30 seconds from what it used to take 71-140 seconds with this enhancement sounds reasonable to me. As there's not so much room left for us from what the framework can offer us to help around the performance, I think it's the best that we can do.

Thank you for updating the `p` to `page`!